x-pack/plugins/console_extensions/server/plugin.ts

```diff
@@ -3,30 +3,90 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import { join } from 'path';
-import { CoreSetup, Logger, Plugin, PluginInitializerContext } from 'kibana/server';
+import {
+  CoreSetup,
+  CoreStart,
+  Logger,
+  Plugin,
+  PluginInitializerContext,
+  SavedObjectsClientContract,
+  SavedObjectsClient,
+} from '../../../../src/core/server';
+
+import { APP } from '../common/constants';
 
 import { ConsoleSetup } from '../../../../src/plugins/console/server';
+import { PluginSetupContract as FeatureSetup } from '../../features/server';
+import { SecurityPluginSetup } from '../../security/server';
+
+import { installExtensions } from './spec/install_extensions';
 
-import { processors } from './spec/ingest/index';
+import { registerTextObjectsRoutes } from './routes/api/text_object';
 
 interface SetupDependencies {
   console: ConsoleSetup;
+  features: FeatureSetup;
+  security: SecurityPluginSetup;
 }
 
 export class ConsoleExtensionsServerPlugin implements Plugin<void, void, SetupDependencies> {
-  log: Logger;
+  private internalSavedObjectsClient: SavedObjectsClientContract | null = null;
+
+  private readonly log: Logger;
+
   constructor(private readonly ctx: PluginInitializerContext) {
     this.log = this.ctx.logger.get();
   }
 
   setup(
-    core: CoreSetup,
-    { console: { addProcessorDefinition, addExtensionSpecFilePath } }: SetupDependencies
+    { http, getStartServices }: CoreSetup,
+    { console, features, security: { authc } }: SetupDependencies
   ) {
-    addExtensionSpecFilePath(join(__dirname, 'spec/'));
-    processors.forEach(processor => addProcessorDefinition(processor));
-    this.log.debug('Installed console autocomplete extensions.');
+    // We register this feature explicitly without direct access to Saved Objects
+    // per docs/developer/plugin/development-plugin-feature-registration.asciidoc
+    features.registerFeature({
+      id: APP.id,
+      name: APP.name,
+      app: [],
+      privileges: {
+        all: {
+          savedObject: {
+            all: [],
+            read: [],
+          },
+          ui: [],
+        },
+        read: {
+          savedObject: {
+            all: [],
+            read: [],
+          },
+          ui: [],
+        },
+      },
+    });
+
+    getStartServices().then(([{ savedObjects }]) => {
+      this.internalSavedObjectsClient = new SavedObjectsClient(
```

Comment:
The name of "internalSavedObjectsClient" is a misnomer, it's technically an instance of the `SavedObjectsRepository`. Using the `SavedObjectsRepository` skips the saved-object authorization which the security plugin performs, which I believe you want; however, it also skips the space-based restrictions which I'm not sure is intentional or not.

Reply:
I'm not sure, for Console, that it makes sense to have your text buffers scoped by space. At least because it does not work that way at the moment and that is the behaviour we will have to offer in OSS.